lesson18/script.js

```diff
@@ -0,0 +1,186 @@
+const slider = document.getElementById("js-slider");
+
+const createAttributedElements = ({tag,attrObj,str}) =>{
+    const element = document.createElement(tag);
+    Object.keys(attrObj).forEach((attribute) => {
+        element.setAttribute(attribute, attrObj[attribute]);
+    });
+    if(str !== undefined) element.textContent = str;
+    return element;
+}
+
+const createErrorMessage = (error) => {
+    const errMsg = createAttributedElements({tag:"p",attrObj:{class:"error-msg"},str:error});
+    
+    slider.appendChild(errMsg);
+}
+
+const showLoading = () => {
+    const loadingPlace = createAttributedElements({tag:"div",attrObj:{id:"js-loading",class:"loading"}});
+    const loadingImage = createAttributedElements({tag:"img",attrObj:{src:"img/loading-circle.gif",alt:"ローディング"}});
+    
+    slider.appendChild(loadingPlace).appendChild(loadingImage);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (URL) => {
+    const response = await fetch(URL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage('現在、サーバーの通信が壊れています');
+    }
+    const json = await response.json();
+    const data = json.data;
+    return data;
```

Comment:
nits
```suggestion
    return json.data;
```

Reply:
Thank you for your advice!
I fixed!
https://github.com/nakagawamai/js-lesson/pull/20/commits/64497a90a847e925d1766898401a5af9a129e0b9